test-data/unit/check-selftype.test

```diff
@@ -351,21 +351,99 @@ class E:
     def __init_subclass__(cls) -> None:
         reveal_type(cls)  # E: Revealed type is 'def () -> __main__.E'
 
-[case testSelfTypeProperty]
-from typing import TypeVar
+[case testSelfTypePropertyUnion]
+from typing import Union
+class A:
+    @property
+    def f(self: A) -> int: pass
 
-T = TypeVar('T', bound='A')
+class B:
+    @property
+    def f(self: B) -> int: pass
+x: Union[A, B]
+reveal_type(x.f)  # E: Revealed type is 'builtins.int'
+
+[builtins fixtures/property.pyi]
 
+[case testSelfTypeProperSupertypeAttribute]
+from typing import Callable, TypeVar
+T = TypeVar('T', bound=object)
```

Comment:
`bound=object` is redundant -- this is equivalent to `T = TypeVar('T')`. It would be nice to have a test case with a non-`object` bound.

Reply:
I know it's redundant, I just wanted to be clear about the intention of the test. But yes, I should add a non-object bound